tests/integration_tests/scheduler/test_openpbs_driver.py

```diff
@@ -96,6 +99,29 @@ async def finished(iens, returncode, aborted):
     assert aborted_called
 
 
+@pytest.mark.timeout(10)
+async def test_keep_qsub_output_option(tmp_path, caplog):
+    driver = OpenPBSDriver(keep_qsub_output=True)
+    with caplog.at_level(logging.DEBUG):
+        await driver.submit(0, "sleep 2")
+    path_regex = re.compile(r"(/[^/\s]+)+")
+    job_id_regex = re.compile(r"(\w+\.\w+)")
+
+    paths_found = [
```

Comment:
I don't think asserting on log statements will suffice. What if someone changes the code but not the log message? Then the test will still pass, but the functionality will be broken.

Reply:
I am not sure how to test this. Should I mock the random function, and assert that the expected path is created and that the file-content is what I expect?

Reply:
You want to assert the presence of some specifically named files (the job id is typically included in the filename). You can have a job sent to the cluster that echoes a magic string to stdout and another magic string to stderr, and then you assert this is present. You will need to figure out where the real qsub puts these files (I think it is eager to pollute `$HOME` 😢 ) and then get the mocked qsub to do the same.